docs/src/content/docs/cookbook/dexes/dedust.mdx

```diff
@@ -7,8 +7,434 @@ sidebar:
+}
+```
+
+### Swap Toncoin for any Jetton
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap
+message(0xea06185d) NativeSwap {
+    // Unique identifier used to trace transactions across multiple contracts
+    // Defaults to 0, which means we don't mark messages to trace their chains
+    queryId: Int as uint64 = 0;
+
+    // Toncoin amount for the swap
+    amount: Int as coins;
+
+    // Inlined fields of SwapStep Struct
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: SwapStep? = null;
+
+    // Set of parameters relevant for the whole swap
+    swapParams: SwapParams;
+}
+
+// Let's say `swapAmount` is `ton("0.1")`, which is 10000000 nanoToncoins
+fun swapToncoinForUSDT(swapAmount: Int) {
+    send(SendParameters{
+        // Address of TON vault to send the message to
+        to: address("EQDa4VOnTYlLvDJ0gZjNYm5PXfSmmtL6Vs6A_CZEtXCNICq_"),
+        // Amount to swap plus a trade fee
+        value: swapAmount + ton("0.2"),
+        body: NativeSwap{
+            amount: swapAmount,
+            // Address of the swap pool, which is the TON/USDT pair in this case
+            poolAddress: address("EQCxE6mUtQJKFnGfaROTKOt1lZbDiiX1kCixRv7Nw2Id_sDs"),
+            // Set of parameters relevant for the whole swap
+            swapParams: SwapParams{}, // use defaults
+        }.toCell(),
+    });
+}
+
+//
+// Helper Structs described earlier on this page
+//
+
+struct SwapStep {
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: Cell?;
+}
+
+struct SwapParams {
+    deadline: Int as uint32 = 0;
+    recipientAddress: Address? = null;
+    referralAddress: Address? = null;
+    fulfillPayload: Cell? = null;
+    rejectPayload: Cell? = null;
+}
+```
+
+### Swap a Jetton for another Jetton or Toncoin
+
+```tact
+/// https://docs.dedust.io/reference/tlb-schemes#message-swap-1
+message(0xe3a0d482) JettonSwapPayload {
+    // Inlined fields of SwapStep Struct
+    poolAddress: Address;
+    kind: Int as uint1 = 0;
+    limit: Int as coins = 0;
+    nextStep: SwapStep? = null;
+
+    // Set of parameters relevant for the whole swap
+    swapParams: SwapParams;
+}
+
+/// NOTE: To calculate and provide Jetton wallet address for the target user,
+///       make sure to check links after this code snippet
+fun swapJetton(targetJettonWalletAddress: Address) {
+    send(SendParameters{
+        to: targetJettonWalletAddress,
+        value: ton("0.3"),
+        body: JettonTransfer{
+            // Unique identifier used to trace transactions across multiple contracts
+            // Set to 0, which means we don't mark messages to trace their chains
+            queryId: 0,
+            // Jetton amount for the swap
+            amount: 10, // NOTE: change to yours
+            // Address of the Jetton vault to the send message to
+            destination: address("EQDa4VOnTYlLvDJ0gZjNYm5PXfSmmtL6Vs6A_CZEtXCNICq_"),
```

Comment:
Maybe just a typo, and user should replace it anyway, but it is Ton dedust Vault address, not Jetton Vault. Address of Usdt dedust vault is `EQAYqo4u7VF0fa4DPAebk4g9lBytj2VFny7pzXR0trjtXQaO`

Reply:
Hmm, can't find it in the list of pools. And [checking latest trades](https://docs.dedust.io/reference/getting-pools-latest-trades) returns nothing for it. Are there other ways to see pools conveniently?

Reply:
i'v used https://docs.dedust.io/docs/swaps#step-2-find-the-pool-ton-scale to find it

Reply:
could be a missprint, will double check it

Reply:
Would be nice to put the link and a bit explanation how to discover this info into the article